src/Backend/OnboardingInsuranceAPI/Areas/Insurance/GetContribution.cs

```diff
@@ -0,0 +1,41 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using OnboardingInsuranceAPI.Enums;
+using OnboardingInsuranceAPI.ErrorHandling;
+using OnboardingInsuranceAPI.Services;
+
+namespace OnboardingInsuranceAPI.Areas.Insurance;
+
+public class GetContribution : IHandler
+{
+    private readonly DataContext _context;
+
+    public GetContribution(DataContext context)
+    {
+        _context = context;
+    }
+
+    public async Task<ContributionData> Handle(ContributionData requestedData, string pid)
+    {
+        if(string.IsNullOrEmpty(pid))
+            throw new ApiException(ErrorCode.InvalidQueryParameters);
+
+        var birthDate = _context.Users.FirstOrDefault(u => u.Pid == pid).Birthdate;
+        if(string.IsNullOrEmpty(birthDate))
+            throw new ApiException(ErrorCode.NotFound);
+
+        var age = DateTime.Now.Year - DateTime.Parse(birthDate).Year;
+        var carProductionYear = DateTime.Parse(requestedData.YearOfProduction).Year;
+        var carTypeContribution = Convert.ToInt32(CarTypesContribution.FromString(requestedData.CarType));
```

Comment:
As i said before. Use straight conversion from string to Enum when deserializing. So you don't have to this nasty conversion everywhere. Also enums can be converted to int just "(int)"
![image](https://user-images.githubusercontent.com/89402322/184917097-d84c3a58-69be-4ccb-bb22-7429c5fb0003.png)

Reply:
https://stackoverflow.com/questions/2441290/javascriptserializer-json-serialization-of-enum-as-string?answertab=trending#tab-top